cloud_foundry_api/datadog_checks/cloud_foundry_api/cloud_foundry_api.py

```diff
@@ -0,0 +1,143 @@
+from datadog_checks.base.types import Event
+
+
+class CloudFoundryApiCheck(AgentCheck):
+    SOURCE_TYPE_NAME = "CloudFoundry"
+    __NAMESPACE__ = "cloud_foundry_api"
+
+    def __init__(self, name, init_config, instances):
+        super(CloudFoundryApiCheck, self).__init__(name, init_config, instances)
+
+        self._api_url = self.instance.get("api_url")
+        self._uaa_url = self.instance.get("uaa_url")
+        self._client_id = self.instance.get("client_id")
+        self._client_secret = self.instance.get("client_id")
+        self._api_version = self.instance.get("api_version", "v3")
+        self._event_filter = ",".join(self.instance.get("event_filter", []))
+        self._tags = self.instance.get("tags", [])
+        self._per_page = self.instance.get("results_per_page", 2000)
+        self._last_event_guid = ""
+        self._last_event_ts = 0
+        self._oauth_token = None
+        self._token_expiration = 0
+
+    def get_oauth_token(self):
+        # type: () -> str
+        if self._oauth_token is not None and self._token_expiration > int(time.time()):
+            return
+        self.log.info("Refreshing access token")
+        res = self.http.get(
+            f"{self._uaa_url}/oauth/token",
+            auth=(self._client_id, self._client_secret),
+            params={"grant_type": "client_credentials"},
+        )
+        try:
+            res.raise_for_status()
+            payload = res.json()
+            if "access_token" in payload and "expires_in" in payload:
+                self._oauth_token = payload["access_token"]
+                self._token_expiration = int(time.time()) + payload["expires_in"]
+                return
+            self.log.error("Could not find access token and expiration in payload: %s", payload)
+        except HTTPError as e:
+            self.log.error("Error authenticating to the UAA server: %s - %s", e, res.text)
+            raise
+        except ValueError as e:
+            self.log.error("Error decoding response from the UAA server: %s - %s", e, res.text)
+            raise
+
+    def get_events_v2(self):
+        events = {}
+        scroll = True
+        params = {
+            "q": f"type IN {self._event_filter}",
+            "results-per-page": min(self._per_page, 100),
+            "order-direction": "desc",
+            "order-by": "timestamp",
+        }
+        headers = {"Authorization": f"Bearer {self._oauth_token}"}
+        res = self.http.get(f"{self._api_url}/v2/events", params=params, headers=headers)
+        while scroll:
+            try:
+                res.raise_for_status()
+                payload = res.json()
+            except HTTPError as e:
+                self.log.error("Error querying list of events: %s - %s", e, res.text)
+                break
+            except ValueError as e:
+                self.log.error("Error decoding response from the UAA server: %s - %s", e, res.text)
+                break
+            # Collect events
```

Comment:
Let's add comment around here to display how the payload look like and have reference links to api documentation for easy access.